src/ucp/stream/stream_recv.c

```diff
@@ -213,12 +213,18 @@ ucp_stream_recv_request_init(ucp_request_t *req, void *buffer, size_t count,
 
     req->recv.buffer   = buffer;
     req->recv.datatype = datatype;
-    req->recv.length   = ucp_dt_length(datatype, count, buffer,
-                                       &req->recv.state);
+    req->recv.length   = length;
 
     ucp_request_recv_state_init(req, buffer, datatype, count);
 }
 
+static UCS_F_ALWAYS_INLINE int
+ucp_stream_recv_nb_is_inplace(ucp_ep_ext_stream_t *ep, size_t dt_length)
+{
+    return (ep->flags & UCP_EP_STREAM_FLAG_HAS_DATA) &&
+            (ucp_stream_recv_data_nb_internal(ep, 0)->length >= dt_length);
```

Comment:
need to remove ucp_stream_recv_data_nb_internal() function, since it just calls one of 2 functions according to a boolean parameter it gets ("dequeue"), and call ucp_stream_rdesc_dequeue()/ucs_queue_head_elem_non_empty() directly instead

Reply:
Yes, but this is single place where i touch the queue for `data`, if remove it, then it will be harder to understand where this queue is used for `data` or `requests`. Second parameter is always constant, so it should be optimised by compiler.

Reply:
will rename to accessor `ucp_stream_recv_get_data`